homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +75,46 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+@TYPES.register('BinarySensor')
+class BinarySensor(HomeAccessory):
+    """Generate a BinarySensor accessory as binary sensor."""
+
+    def __init__(self, hass, state, **kwargs):
+        """Initialize a BinarySensor accessory object."""
+        entity_id = state.entity_id
```

Comment:
Isn't necessary if `entity_id` is passed to constructor.

Reply:
Fixed